tests/model_fields/test_jsonfield.py

```diff
@@ -0,0 +1,72 @@
+import unittest
+import uuid
+
+from django.db import connection, transaction
+from django.test import TestCase
+
+from .models import JSONCustomEncoderDecoderModel, JSONModel
+
+
+class JSONFieldTests(TestCase):
+
+    @unittest.skipIf(
+        connection.vendor == 'oracle',
+        "Oracle Database doesn't support scalar JSON values."
+    )
+    def test_scalar_values(self):
+        obj = JSONModel.objects.create(value=True)
+        obj = JSONModel.objects.get(id=obj.id)
+        self.assertEqual(obj.value, True)
+
+        obj = JSONModel.objects.create(value=False)
+        obj = JSONModel.objects.get(id=obj.id)
+        self.assertEqual(obj.value, False)
+
+        obj = JSONModel.objects.create(value=123456)
+        obj = JSONModel.objects.get(id=obj.id)
+        self.assertEqual(obj.value, 123456)
+
+        obj = JSONModel.objects.create(value=1234.56)
+        obj = JSONModel.objects.get(id=obj.id)
+        self.assertEqual(obj.value, 1234.56)
+
+        obj = JSONModel.objects.create(value='A string')
+        obj = JSONModel.objects.get(id=obj.id)
+        self.assertEqual(obj.value, 'A string')
```

Comment:
I'd split this into multiple tests, one per value. See the AAA pattern: https://jamescooke.info/cleaner-unit-testing-with-the-arrange-act-assert-pattern.html

Also you can test the error case(s) on Oracle for trying to store scalar values

Reply:
Done (I think?). Not sure if I'm doing it correctly, though.